src/react-components/room/ChatSidebarContainer.js

```diff
@@ -0,0 +1,245 @@
+    message => {
+      if (messageDispatch) {
+        messageDispatch.dispatch(message);
+      }
+    },
+    [messageDispatch]
+  );
+
+  const setMessagesRead = useCallback(
+    () => {
+      setUnreadMessages(false);
+    },
+    [setUnreadMessages]
+  );
+
+  return (
+    <ChatContext.Provider value={{ messageGroups, unreadMessages, sendMessage, setMessagesRead }}>
+      {children}
+    </ChatContext.Provider>
+  );
+}
+
+ChatContextProvider.propTypes = {
+  children: PropTypes.node,
+  messageDispatch: PropTypes.object
+};
+
+export function ChatSidebarContainer({ canSpawnMessages, onUploadFile, discordBridges, occupantCount, onClose }) {
+  const { messageGroups, sendMessage, setMessagesRead } = useContext(ChatContext);
+  const [onScrollList, listRef, scrolledToBottom] = useMaintainScrollPosition(messageGroups);
+  const [message, setMessage] = useState("");
+  const onKeyDown = useCallback(
+    e => {
+      if (e.key === "Enter" && !e.shiftKey) {
+        e.preventDefault();
+        sendMessage(e.target.value);
+        setMessage("");
+      }
+    },
+    [sendMessage, setMessage]
+  );
+  const onSpawnMessage = () => {
+    spawnChatMessage(message);
+    setMessage("");
+  };
+
+  const onUploadAttachments = e => {
+    // TODO: Right now there's no way to upload files to the chat only.
+    // When we add the place menu whcih will have an explicit button for uploading files,
+    // should we make this attach button only upload to chat?
+    for (const file of e.target.files) {
+      onUploadFile(file);
+    }
+  };
+
+  useEffect(
+    () => {
+      if (scrolledToBottom) {
+        setMessagesRead();
+      }
+    },
+    [messageGroups, scrolledToBottom, setMessagesRead]
+  );
+
+  // TODO: Add i18n for this placeholder
+  const discordSnippet = discordBridges.map(ch => "#" + ch).join(", ");
+  const occupantSnippet = `${occupantCount - 1} other${occupantCount > 2 ? "s" : ""}`;
+  const placeholder =
+    occupantCount <= 1
+      ? "Nobody is here yet..."
+      : discordBridges.length
+        ? `Send message to ${occupantSnippet} and ${discordSnippet}...`
+        : `Send message to ${occupantSnippet}...`;
+
+  return (
+    <ChatSidebar onClose={onClose}>
+      <ChatMessageList ref={listRef} onScroll={onScrollList}>
+        {messageGroups.map(({ id, systemMessage, ...rest }) => {
+          if (systemMessage) {
+            return <SystemMessage key={id} {...rest} />;
+          } else {
+            return <ChatMessageGroup key={id} {...rest} />;
+          }
+        })}
+      </ChatMessageList>
+      <ChatInput
+        id="chat-input"
+        onKeyDown={onKeyDown}
+        onChange={e => setMessage(e.target.value)}
+        placeholder={placeholder}
+        value={message}
+        afterInput={
+          canSpawnMessages && (
+            <>
+              <MesssageAttachmentButton onChange={onUploadAttachments} />
+              <SpawnMessageButton onClick={onSpawnMessage} />
```

Comment:
Related to the above point - the behavior of this button is a bit surprising. Feels like there a "send to chat" button missing. (I know you send the message to chat by pressing enter, but this button _seems_ like it would do the same thing.)
If chat messages had a magic wand button show up to the left on hover, and that let you spawn that message into the room, then we might be able to have this magic wand `SpawnMessageButton` show up on the left side of the chat input in a similar fashion (possibly only when there is something written in the message or only on hover).

Reply:
This button seems surprisingly "active"/responsive when there is no current message. Perhaps add a `disactivated` state until the user writes something

Reply:
I added disabled style. I'll respond to the rest in the main thread.